incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -252,6 +258,14 @@ func (r *ObjectReconciler) propagate(ctx context.Context, log logr.Logger, inst
 	return nil
 }
 
+// isObjectFinalizerEmpty checks if object has nil value in finalizer or not
+func isObjectFinalizerEmpty(inst *unstructured.Unstructured) bool {
+	if inst != nil && inst.GetFinalizers() != nil {
```

Comment:
You should never compare a Go slice to `nil` to check to see whether it's empty, because it can be allocated (non-nil) but still have zero objects in it. Instead, say:

`len(inst.GetFinalizers()) != 0`

since `len()` works for both `nil` and non-nil slices.